submission/submission-webapp/src/itest/kotlin/ac/uk/ebi/biostd/itest/assertions/AllInOneSubmissionHelper.kt

```diff
@@ -0,0 +1,49 @@
+package ac.uk.ebi.biostd.itest.assertions
+
+import ac.uk.ebi.biostd.client.integration.web.BioWebClient
+import ac.uk.ebi.biostd.itest.factory.allInOneSubmission
+import ac.uk.ebi.biostd.itest.factory.assertAllInOneSubmissionJson
+import ac.uk.ebi.biostd.itest.factory.assertAllInOneSubmissionTsv
+import ac.uk.ebi.biostd.itest.factory.assertAllInOneSubmissionXml
+import ac.uk.ebi.biostd.persistence.service.SubmissionRepository
+import ebi.ac.uk.asserts.assertThat
+import ebi.ac.uk.model.ExtendedSubmission
+import io.github.glytching.junit.extension.folder.TemporaryFolder
+import org.assertj.core.api.Assertions.assertThat
+import java.nio.file.Paths
+
+internal class AllInOneSubmissionHelper(
+    private val basePath: String,
+    private val submissionRepository: SubmissionRepository
+) {
+    internal fun createAllInOneSubmissionFiles(webClient: BioWebClient, tempFolder: TemporaryFolder) {
+        tempFolder.createDirectory("Folder1")
+        tempFolder.createDirectory("Folder1/Folder2")
+
+        webClient.uploadFiles(listOf(tempFolder.createFile("DataFile1.txt"), tempFolder.createFile("DataFile2.txt")))
+        webClient.uploadFiles(listOf(tempFolder.createFile("Folder1/DataFile3.txt")), "Folder1")
+        webClient.uploadFiles(listOf(tempFolder.createFile("Folder1/Folder2/DataFile4.txt")), "Folder1/Folder2")
+    }
+
+    internal fun assertSavedSubmission(accNo: String) {
+        val extendedSubmission = submissionRepository.getExtendedByAccNo(accNo)
+        assertThat(extendedSubmission.asSubmission()).isEqualTo(allInOneSubmission(accNo))
```

Comment:
I was cour equals and hascode implementation and I think we need to change them, is ok to use them to compare object basic properties as title, attributes, url but not for nested objects as this is making hibernate load the full entity three (making lazy loading basically useless) . That refactor is out of the scope of this issue but if we trust equals latter we will need to re introduce assertions